sklearn/impute/tests/test_impute.py

```diff
@@ -351,6 +351,21 @@ def test_imputation_most_frequent_pandas(dtype):
     assert_array_equal(X_trans, X_true)
 
 
+def test_imputation_most_frequent_string_list():
+    X = [['a', 'b'],
+         ['c', np.nan]]
+
+    X_true = np.array([
+        ['a', 'b'],
+        ['c', 'b']
+    ], dtype=object)
+
+    imputer = SimpleImputer(strategy="most_frequent")
```

Comment:
```suggestion
@pytest.mark.parametrize(
    'strategy, expected',
    [('most_frequent', 'b'), ('constant', 'missing_value')]
)
def test_imputation_most_frequent_string_list(strategy, expected):
    X = [['a', 'b'],
         ['c', np.nan]]

    X_true = np.array([
        ['a', 'b'],
        ['c', expected]
    ], dtype=object)

    imputer = SimpleImputer(strategy=strategy)
```

and remove the other test.

Reply:
@rth Thank you! I followed your suggestion:)

In addition, little modifications are done in following commits
* `test_imputation_constant_string_list` is removed (because it is done by parameterized new unittest)
* The unittest name is changed to `test_simple_imputation_string_list ` (which is combined  `test_imputation_most_frequent_string_list` and `test_imputation_constant_string_list `)